src/WebJobs.Script.WebHost/WebScriptHostExceptionHandler.cs

```diff
@@ -29,6 +29,13 @@ public async Task OnTimeoutExceptionAsync(ExceptionDispatchInfo exceptionInfo, T
         {
             FunctionTimeoutException timeoutException = exceptionInfo.SourceException as FunctionTimeoutException;
 
+            // this seems to happen when the worker channel is already shutting down. Ex. One timeout is being handled and another comes in during shutdown.
+            if (timeoutException is null)
```

Comment:
If it's not `FunctionTimeoutException`, what exception the caller is sending to this method?

Reply:
Yeah this seems odd, the method is `OnTimeoutExceptionAsync`, but we are now checking if it was called for a non-timeout? Additionally, the log here just assumes that if `timeoutException is null`, then the channel _must_ be shutting down? Will that always be true? How future proof is that assumption?